cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -1186,9 +1186,9 @@ TYPED_TEST(TestStringKernels, BinaryJoin) {
   auto expected =
       ArrayFromJSON(this->type(), R"(["a--bb--ccc", "", null, "dd", null, "ff--"])");
   CheckScalarBinary("binary_join", ArrayFromJSON(list(this->type()), list_json),
-                    separator, expected);
+                    static_cast<Datum>(separator), expected);
   CheckScalarBinary("binary_join", ArrayFromJSON(large_list(this->type()), list_json),
-                    separator, expected);
+                    static_cast<Datum>(separator), expected);
```

Comment:
Ah, this isn't a cast, but is instead invoking the implicit conversion ([case (1) here](https://en.cppreference.com/w/cpp/language/static_cast)). Can we write it as just `Datum(separator)` to make that clear?

Reply:
Done